src/Checkbox.tsx

```diff
@@ -0,0 +1,74 @@
+import styled from 'styled-components'
+import React, {InputHTMLAttributes, ReactElement, useLayoutEffect, useRef} from 'react'
+import sx, {SxProp} from './sx'
+
+export type CheckboxProps = {
+  /**
+   * Apply indeterminate visual appearance to the checkbox
+   */
+  indeterminate?: boolean
+  /**
+   * Apply inactive visual appearance to the checkbox
+   */
+  disabled?: boolean
+  /**
+   * Forward a ref to the underlying input element
+   */
+  ref?: React.RefObject<HTMLInputElement>
+  /**
+   * Indicates whether the checkbox must be checked
+   */
+  required?: boolean
+
+  /**
+   * Indicates whether the checkbox validation state
+   */
+  validationStatus?: 'error' | 'success' // TODO: hoist to Validation typings
+} & InputHTMLAttributes<HTMLInputElement> &
+  SxProp
+
+const StyledCheckbox = styled.input`
+  cursor: pointer;
+
+  ${props => props.disabled && `cursor: not-allowed;`}
+
+  ${sx}
+`
+
+/**
+ * An accessible, native checkbox component
+ */
+const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
+  (
+    {checked, indeterminate, disabled, sx: sxProp, required, validationStatus, ...rest}: CheckboxProps,
+    ref
+  ): ReactElement => {
+    const checkboxRef = useRef<HTMLInputElement>(null)
+
+    useLayoutEffect(() => {
+      if (checkboxRef.current) {
+        checkboxRef.current.indeterminate = indeterminate || false
+      }
+    }, [indeterminate, checked])
+
+    return (
+      <StyledCheckbox
+        type="checkbox"
+        disabled={disabled}
+        aria-disabled={disabled ? 'true' : 'false'}
+        ref={ref || checkboxRef}
+        checked={indeterminate ? false : checked}
+        aria-checked={indeterminate ? 'mixed' : checked ? 'true' : 'false'}
+        sx={{...sxProp}}
```

Comment:
Why do we need to spread `...sxProp` here?
```suggestion
        sx={sxProp}
```

Reply:
Fixed.